### ISSUE TITLE
[🐛] When running with NODE_ENV=unsafe any unhandled exception is crashing the server

### ISSUE THREAD

**Issue by harisec** [juice-shop/juice-shop#2676]
# :bug: Bug report

## Description

When starting juice shop using docker and enabling potentially dangerous challenges via **NODE_ENV=unsafe** any unhandled exception in the application will crash the application. 

IMO, it would be better to adjust the command from the documentation that starts the application via docker to threat unhandled rejections as a warning instead of crashing the application:

For example, the following command:
```
docker run  -e NODE_ENV=unsafe  -e NODE_OPTIONS="--unhandled-rejections=warn"  --rm -p 127.0.0.1:3000:3000 bkimminich/juice-shop
```

will have the juice shop application keep running after an unhandled exception.

I've added the **NODE_OPTIONS="--unhandled-rejections=warn"**. The **--unhandled-rejections=warn** argument instructs Node.js to show a console warning for an unhandled promise rejection instead of crashing the application. 

The current recommended command from the documentation will crash the application:

```
docker run  -e NODE_ENV=unsafe  --rm -p 127.0.0.1:3000:3000 bkimminich/juice-shop
```

### Is this a regression?

No

## :microscope: Minimal Reproduction

Start juice shop with the following command:
```
docker run -e NODE_ENV=unsafe -p 127.0.0.1:3000:3000 bkimminich/juice-shop
```
1. Visit the url `http://127.0.0.1:3000/`, login with `demo:demo`
2. Go to the profile page: `http://127.0.0.1:3000/profile`

<img width="959" height="491" alt="Image" src="https://github.com/user-attachments/assets/7873d2be-81f0-4d6c-a6f1-b6087661641f" />
 
3. Enter the following value for the username: ```<%={{={@{#{${dfb}}%>```
4. The application will crash, docker will terminate

<img width="958" height="458" alt="Image" src="https://github.com/user-attachments/assets/568dc788-ab2d-423c-8063-917d4b6cdcd6" />

## :fire: Exception or Error

<pre><code>
docker run -e NODE_ENV=unsafe -p 127.0.0.1:3000:3000 bkimminich/juice-shop
info: Detected Node.js version v22.16.0 (OK)
info: Detected OS linux (OK)
info: Detected CPU x64 (OK)
info: Configuration unsafe validated (OK)
info: Entity models 19 of 19 are initialized (OK)
info: Required file server.js is present (OK)
info: Required file index.html is present (OK)
info: Required file styles.css is present (OK)
info: Required file main.js is present (OK)
info: Required file tutorial.js is present (OK)
info: Required file runtime.js is present (OK)
info: Required file vendor.js is present (OK)
info: Port 3000 is available (OK)
info: Chatbot training data botDefaultTrainingData.json validated (OK)
info: Domain https://www.alchemy.com/ is reachable (OK)
info: Server listening on port 3000
/juice-shop/node_modules/pug-error/lib/index.js:34
    var err = new Error(fullMessage);
              ^

Error: Pug:45:96
    43|                  .mdl-cell.mdl-cell--6-col-desktop.mdl-cell--12-col-tablet.mdl-cell--12-col-phone
    44|                     img.img-rounded(src=profileImage, alt='profile picture', width='90%', height='236', style='margin-right: 5%; margin-left: 5%;')
  > 45|                     p(style='margin-top:8%; color: #FFFFFF; text-align: center;') \<%={{={@{#{${dfb}}%>
-------------------------------------------------------------------------------------------------------^
    46|                     form(action='./profile/image/file' , style='margin-top:10%; width: 90%; margin-right: auto; margin-left: auto;', method='post', enctype='multipart/form-data')
    47|                       .form-group
    48|                         label(for='picture', style='color: #FFFFFF; font-size: 12px;') File Upload:

Syntax Error: Unexpected token
    at makeError (/juice-shop/node_modules/pug-error/lib/index.js:34:15)
    at Lexer.error (/juice-shop/node_modules/pug-lexer/index.js:62:15)
    at Lexer.assertExpression (/juice-shop/node_modules/pug-lexer/index.js:96:12)
    at Lexer.addText (/juice-shop/node_modules/pug-lexer/index.js:627:12)
    at Lexer.text (/juice-shop/node_modules/pug-lexer/index.js:653:12)
    at Lexer.callLexerFunction (/juice-shop/node_modules/pug-lexer/index.js:1647:23)
    at Lexer.advance (/juice-shop/node_modules/pug-lexer/index.js:1689:12)
    at Lexer.callLexerFunction (/juice-shop/node_modules/pug-lexer/index.js:1647:23)
    at Lexer.getTokens (/juice-shop/node_modules/pug-lexer/index.js:1706:12)
    at lex (/juice-shop/node_modules/pug-lexer/index.js:12:42)
    at Object.lex (/juice-shop/node_modules/pug/lib/index.js:104:9)
    at Function.loadString [as string] (/juice-shop/node_modules/pug-load/index.js:53:24)
    at compileBody (/juice-shop/node_modules/pug/lib/index.js:82:18)
    at exports.compile (/juice-shop/node_modules/pug/lib/index.js:269:16)
    at /juice-shop/build/routes/userProfile.js:106:34 {
  code: 'PUG:SYNTAX_ERROR',
  msg: 'Syntax Error: Unexpected token',
  line: 45,
  column: 96,
  filename: undefined,
  src: 'doctype html\n' +
    "html(lang='en')\n" +
    '    head\n' +
    '        title OWASP Juice Shop\n' +
    "        meta(charset='utf-8')\n" +
    "        meta(name='description', content='')\n" +
    "        meta(name='keywords', content='')\n" +
    "        meta(name='viewport' content='width=device-width, initial-scale=1.0')\n" +
    "        link(rel='icon', type='image/x-icon', href='./assets/public/favicon_js.ico')\n" +
    "        link(rel='stylesheet', href='https://code.getmdl.io/1.3.0/material.min.css')\n" +
    "        link(rel='stylesheet', href='https://fonts.googleapis.com/icon?family=Material+Icons')\n" +
    "        link(rel='stylesheet', href='./assets/public/css/userProfile.css', type='text/css')\n" +
    "        link(rel='stylesheet', href='http://fonts.googleapis.com/css?family=Roboto:300,400,500,700', type='text/css')\n" +
    "        script(src='//ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js')\n" +
    "        script(src='https://code.getmdl.io/1.3.0/material.min.js')\n" +
    '        style.\n' +
    '            .mdl-textfield__input {\n' +
    '            border-bottom: 1px solid #FFFFFF !important;\n' +
    '            font-size: 13px !important;\n' +
    '            }\n' +
    "    body(style='background: #303030;color:#FFFFFF;')\n" +
    '        .mdl-layout.mdl-js-layout.mdl-layout--fixed-header\n' +
    "          header.mdl-layout__header.mdl-shadow--8dp(style= 'background: #546E7A; height: auto; min-width: 100%; padding-bottom: 5px; width: 100%;')\n" +
    '            .mdl-layout__header-row\n' +
    "              a(href='./#/' style='color: #FFFFFF; text-decoration:none; margin-left: -50px;')\n" +
    "                i(class='material-icons', style='display: block;margin-bottom: auto;margin-top: auto; margin-right: 10px;') arrow_back\n" +
    "              a(href='./#/' style='color: #FFFFFF; text-decoration:none;')\n" +
    "                span(style='margin-right: 20px;')\n" +
    '                  | Back\n' +
    '              // Logo\n' +
    "              a(href='./#/')\n" +
    "                img(src='assets/public/images/JuiceShop_Logo.png', style='max-height: 60px; width: auto;', alt='OWASP Juice Shop Logo')\n" +
    '              // Title\n' +
    "              a(href='./#/' style='color: #FFFFFF; text-decoration:none;')\n" +
    `                span.mdl-layout-title(style='font: 500 20px/32px Roboto,"Helvetica Neue",sans-serif;')\n` +
    '                  | OWASP Juice Shop\n' +
    '\n' +
    "        main.mdl-layout__content(style=' display: block; margin-left: auto; margin-right:auto;')\n" +
    '          section.section--center.mdl-grid.mdl-grid--no-spacing\n' +
    "            .mdl-card#card.mdl-cell.mdl-cell--12-col.mdl-shadow--8dp(style='height: auto; min-width: 300px;  width: 40%; display: block; margin-left: auto; margin-right:auto; background: #424242; margin-bottom: 50px; margin-top: 110px;')\n" +
    '              .mdl-card__supporting-text.mdl-grid.mdl-grid--no-spacing\n' +
    "                 h1.mdl-cell.mdl-cell--12-col(style='color: #FFFFFF; font-size: 24px; line-height: 32px; margin-top: 16px; margin-bottom: 16px; font-weight: 400;') User Profile\n" +
    '                 .mdl-cell.mdl-cell--6-col-desktop.mdl-cell--12-col-tablet.mdl-cell--12-col-phone\n' +
    "                    img.img-rounded(src=profileImage, alt='profile picture', width='90%', height='236', style='margin-right: 5%; margin-left: 5%;')\n" +
    "                    p(style='margin-top:8%; color: #FFFFFF; text-align: center;') \\<%={{={@{#{${dfb}}%>\n" +
    "                    form(action='./profile/image/file' , style='margin-top:10%; width: 90%; margin-right: auto; margin-left: auto;', method='post', enctype='multipart/form-data')\n" +
    '                      .form-group\n' +
    "                        label(for='picture', style='color: #FFFFFF; font-size: 12px;') File Upload:\n" +
    "                        input#picture(type='file', accept='image/*', name='file', size='150', style='color: #FFFFFF; margin-top: 4px;', aria-label='Input for selecting the profile picture')\n" +
    "                        .mdl-tooltip(for='picture', style='width: 150px; text-align: left;')\n" +
    '                          | &bull; Maximum file size 150Kb\n' +
    '                          br\n' +
    '                          | &bull; All image formats are accepted\n' +
    "                      button.mdl-button.mdl-js-button.mdl-button--raised.mdl-js-ripple-effect(type='submit', style='background-color:#546E7A; color: #FFFFFF; margin-top: 3%; text-transform: capitalize;', aria-label='Button to upload the profile picture') Upload Picture\n" +
    '\n' +
    "                    .breakLine(style='margin-top: 3%; margin-bottom: 3%; width: 90%; margin-right: auto; margin-left: auto;')\n" +
    '                      .line\n' +
    '                        div\n' +
    "                      .textOnLine(style='color: #FFFFFF;') or\n" +
    '                      .line\n' +
    '                        div\n' +
    '\n' +
    "                    form(action='./profile/image/url' , style='margin-top:5%; width: 90%; margin-right: auto; margin-left: auto;', method='post')\n" +
    '                      .form-group\n' +
    "                        .mdl-textfield.mdl-js-textfield.mdl-textfield--floating-label(style='width: 100%;')\n" +
    "                          input#url.form-control.mdl-textfield__input(type='text', name='imageUrl', style='color: #FFFFFF;', placeholder='e.g. https://www.gravatar.com/avatar/fe01ce2a7fbac8fafaed7c982a04e229', aria-label='Text field for the image link')\n" +
    "                          label.mdl-textfield__label(for='url', style='color: #FFFFFF;') Image URL:\n" +
    "                      button(id='submitUrl', type='submit', style='background-color:#546E7A; color: #FFFFFF; margin-top: -10px; text-transform: capitalize;', aria-label='Button to include image from link').mdl-button.mdl-js-button.mdl-button--raised.mdl-js-ripple-effect Link Image\n" +
    "                      p(style='margin-bottom:10%;')\n" +
    '\n' +
    '                 .mdl-cell.mdl-cell--6-col-desktop.mdl-cell--12-col-tablet.mdl-cell--12-col-phone\n' +
    "                   form(action='./profile', method='post', style='width: 90%; margin-right: auto; margin-left: auto;')\n" +
    '                     .form-group\n' +
    "                         .mdl-textfield.mdl-js-textfield.mdl-textfield--floating-label(style='width: 100%; opacity: 0.7')\n" +
    "                            input#email.form-control.mdl-textfield__input(type='email', name='email', value=email, disabled=true, style='color: #FFFFFF;', aria-label='Disabled - Text field for the email')\n" +
    "                            label.mdl-textfield__label(for='email', style='color: #FFFFFF;') Email:\n" +
    '                     .form-group\n' +
    "                         .mdl-textfield.mdl-js-textfield.mdl-textfield--floating-label(style='width: 100%;')\n" +
    "                            input#username.form-control.mdl-textfield__input(type='text', name='username', value=username ,style='color: #FFFFFF;', placeholder='e.g. SuperUser', aria-label='Text field for the username')\n" +
    "                            label.mdl-textfield__label(for='username', style='color: #FFFFFF;') Username:\n" +
    "                     button(id='submit', type='submit', style='background-color:#546E7A; color: #FFFFFF; margin-top: -10px; text-transform: capitalize;', aria-label='Button to save/set the username').mdl-button.mdl-js-button.mdl-button--raised.mdl-js-ripple-effect Set Username\n" +
    "                     p(style='margin-bottom:10%;')",
  toJSON: [Function (anonymous)]
}

Node.js v22.16.0
</code></pre>

## :deciduous_tree: Your Environment

<pre><code>
Running via docker
</code></pre>

**Comment by J12934** [comment]
Hi @harisec 
sorry for the late response,
thanks for the detailed report, can reproduce.
I agree that this shouldn't crash the server, so adding error handling here to route would be good.

i wouldn't want to set the `--unhandled-rejections=warn` flag, I'd rather fix the cases that we have and they are more likely to be noticed if the server crashes if it does.

**Comment by bkimminich** [comment]
Just to clarify the scope/impact: On a locally installed Juice Shop this will also happen in `default` config. On Docker or Heroku you need the `unsafe` config because otherwise the whole vulnerability of the profile username field does not exist.

**Comment by seershan** [comment]
Hello @J12934 ! After reviewing the issue and the recent comments, I'm starting to work on a fix for this now. I will aim to submit a pull request shortly.

**Comment by J12934** [comment]
reopening as i still think this should be fixed and should be somewhat easy to do
